enderio-machines/src/main/java/com/enderio/machines/client/EnchanterScreen.java

```diff
@@ -0,0 +1,55 @@
+package com.enderio.machines.client;
+
+import com.enderio.base.EnderIO;
+import com.enderio.core.client.screen.EIOScreen;
+import com.enderio.core.client.screen.EnumIconWidget;
+import com.enderio.core.common.util.Vector2i;
+import com.enderio.machines.common.menu.EnchanterMenu;
+import com.mojang.blaze3d.vertex.PoseStack;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Inventory;
+
+public class EnchanterScreen extends EIOScreen<EnchanterMenu>{
+    private static final ResourceLocation BG_TEXTURE = new ResourceLocation(EnderIO.DOMAIN, "textures/gui/enchanter.png");
+
+    public EnchanterScreen(EnchanterMenu pMenu, Inventory pPlayerInventory, Component pTitle) {
+        super(pMenu, pPlayerInventory, pTitle);
+    }
+    
+    @Override
+    protected void init() {
+        super.init();
+        addRenderableWidget(new EnumIconWidget<>(this, leftPos + imageWidth - 8 - 12, topPos + 6, () -> menu.getBlockEntity().getRedstoneControl(), control -> menu.getBlockEntity().setRedstoneControl(control)));
+    }
+    
+    @Override
+    protected ResourceLocation getBackgroundImage() {
+        return BG_TEXTURE;
+    }
+
+    @Override
+    protected Vector2i getBackgroundImageSize() {
+        return new Vector2i(176, 166);
+    }
+
+    @Override
+    protected void renderBg(PoseStack pPoseStack, float pPartialTicks, int pMouseX, int pMouseY) {
+        renderGradleWeirdnessBackground(pPoseStack, pPartialTicks, pMouseX, pMouseY);
+    }
+    
+    @Override
+    public void render(PoseStack pPoseStack, int pMouseX, int pMouseY, float pPartialTicks) {
+        super.render(pPoseStack, pMouseX, pMouseY, pPartialTicks);
+        if (this.getMenu().getCurrentCost() < 0) return;
+        int j = 8453920;
```

Comment:
What is j and what is the number, can we use names and literals if possible?

Reply:
Its the same as vanilla, I will change it to "color"